src/main/java/uk/gov/hmcts/cmc/claimstore/services/ccd/CoreCaseDataService.java

```diff
@@ -853,4 +854,50 @@ public void savePaidInFull(Long caseId, PaidInFull paidInFull, String authorisat
             );
         }
     }
+
+    public Claim linkLetterHolder(Long caseId, String letterHolderId) {
+        try {
+            User anonymousCaseWorker = userService.authenticateAnonymousCaseWorker();
+
+            UserDetails userDetails = anonymousCaseWorker.getUserDetails();
+            EventRequestData eventRequestData = eventRequest(LINK_LETTER_HOLDER, userDetails.getId());
+
+            StartEventResponse startEventResponse = startUpdate(
+                anonymousCaseWorker.getAuthorisation(),
+                eventRequestData,
+                caseId,
+                userDetails.isSolicitor() || userDetails.isCaseworker()
+            );
+
+            CCDCase ccdCase = extractCase(startEventResponse.getCaseDetails());
```

Comment:
The letterholderId should be passed as part of event.

Reply:
it's our existing process of conversion from case to claim